src/aps/toolbox/visualize_prob_curves.py

```diff
@@ -0,0 +1,294 @@
+        cum_prob[n, :] = sumprob[:]
+
+    num_val_refined = num_val * refinement_factor
+    refined_cum_prob = np.zeros((num_logs, num_val_refined), dtype=np.float32)
+    z_indices = np.zeros(num_val_refined, dtype=np.int32)
+    zone_log_val_refined = np.zeros(num_val_refined, dtype=np.int32)
+    indx = 0
+    for j in range(num_val):
+        for i in range(refinement_factor):
+            refined_cum_prob[:, indx] = cum_prob[:, j]
+            if zone_log_val[j] >= 0.0:
+                zone_log_val_refined[indx] = zone_log_val[j]
+            else:
+                zone_log_val_refined[indx] = 0
+            indx += 1
+    z_indices = np.arange(0, num_val_refined)
+    z_values = np.arange(0, num_val, 1.0 / refinement_factor)
+
+    return refined_cum_prob, z_indices, z_values, zone_log_val_refined
+
+
+def read_model_file(model_file_name):
+    suffix = Path(model_file_name).suffix.lower().strip('.')
+    if suffix in ['yaml', 'yml']:
+        param_dict = _read_model_file_yml(model_file_name)
+    elif suffix == 'xml':
+        raise ValueError(f'No xml file format implemented for {__name__}   ')
+    else:
+        raise ValueError(f"Model file name: {model_file_name}  must be 'yml' format")
+    return param_dict
+
+
+def _read_model_file_yml(model_file_name):
+    print(f'Read model file: {model_file_name}')
+    spec_all = readYml(model_file_name)
+
+    kw_parent = 'PlotProbLogs'
+    spec = spec_all[kw_parent] if kw_parent in spec_all else None
+    if spec is None:
+        raise ValueError(f'Missing keyword: {kw_parent} ')
+
+    params = {}
+    plot_vpc = get_bool_value(spec, 'PlotEstimatedVPC', False)
+    params['plot_vpc'] = plot_vpc
+    if not plot_vpc:
+        params['debug_level'] = get_int_value(
+            spec, kw_parent, 'DebugLevel', has_default=True, default_value=Debug.OFF
+        )
+        params['grid_model_name'] = get_text_value(spec, kw_parent, 'GridModelName')
+        params['bw_set_name'] = get_text_value(spec, kw_parent, 'BlockedWellSetName')
+        params['zone_log_name'] = get_text_value(spec, kw_parent, 'ZoneLogName')
+        params['max_zone_number'] = get_int_value(spec, kw_parent, 'MaxZoneNumber')
+        params['well_list'] = get_list(spec, kw_parent, 'WellList')
+        params['prob_log_names'] = get_list(spec, kw_parent, 'ProbLogNames')
+    else:
+        params['average_prob_log_prefix'] = get_text_value(
+            spec, kw_parent, 'AverageProbLogPrefix'
+        )
+        params['average_prob_log_file_path'] = get_text_value(
+            spec, kw_parent, 'AverageProbLogFilePath'
+        )
+        params['facies_names'] = get_list(spec, kw_parent, 'Facies')
+        params['zone_name'] = get_text_value(spec, kw_parent, 'ZoneName')
+    return params
+
+
+def read_vpc(params: dict):
+    facies_list = params['facies_names']
+    zone_name = params['zone_name']
+    prefix = params['average_prob_log_prefix']
+    path = params['average_prob_log_file_path']
```

Comment:
```suggestion
def read_vpc(facies_list, zone_name, prefix, path):
```

Consider using explicit argument names

`read_vpc` can still be called with `params` by using `read_vpc(**params)`